### ISSUE TITLE
Un/publishing video track on macOS is broken

### ISSUE THREAD

**Issue by creativecbr** [livekit/client-sdk-swift#331]
**Describe the Bug**

When streaming the display and subsequently shutting it down, the streaming automatically stops. If attempting to unpublish the display track, it returns an error stating that the track either does not exist or is already stopped. However, when the display turns back on, it does not automatically republish. When attempting to publish a newly created track with the same display ID, an error message is received:

```
'The same track couldn't be used, error: Timed out, trying to initialize a new one.'
```

**SDK Version**
I've identified a bug in the main branch of macOS Swift's LiveKit version. This issue did not occur in the released version 1.16.

**iOS/macOS Version**
macOS 12.7.1, macOS 14.2.1.

**Steps to Reproduce**
0. Set the display you want to use for screen sharing as the main display in Settings -> Displays.
1. Publish a video track from your screen; share the screen on macOS (using your sample app, for instance).
2. Turn off the display using the physical button; streaming continues.
3. Attempt to stop the published track using publication - an error is encountered.
4. Turn on the display.
5. Attempt to gather a track from this display and publish it in the room - an error "can't publish an already published track" is received, and the stream is not resumed.

Just changing a video stream setting will resume streaming - for example, logging with "setting layer to," and then streaming starts again.

**Expected Behavior**
Unpublishing a track that is already stopped should work, and publishing the same display track should succeed.

**Comment by pblazej** [comment]
@creativecbr is it still reproducible in the current (2.x) version?

**Comment by pblazej** [comment]
Yes, it is, but the track is "frozen" on the last frame. Let's see what we can do about it.
